plugins/out_stackdriver/stackdriver.c

```diff
@@ -2510,6 +2515,91 @@ static void update_retry_metric(struct flb_stackdriver *ctx,
 }
 #endif
 
+static int extract_map_from_msgpack(msgpack_object *root, char *name, int size, msgpack_object_type object_type ,msgpack_object *val){
+    int i;
+    msgpack_object key;
+    for (i = 0; i < root->via.map.size; i++) {
+        key = root->via.map.ptr[i].key;
+        if (key.type != MSGPACK_OBJECT_STR) {
+                continue;
+        }
+        if(key.via.str.size == size && strncmp(key.via.str.ptr, name, size) == 0){
+            *val = root->via.map.ptr[i].val;
+            if (val->type != object_type) {
+                return -1;
+            }
+            return 0;
+        }
+    }
+    return -1;
+}
+
+static int parse_partial_success_response(struct flb_http_client *c, struct flb_stackdriver *ctx, int *partial_failure){
+    int ret;
+    int root_type;
+    int i;
+    char *buffer;
+    size_t size;
+    size_t off = 0;
+    msgpack_unpacked result;
+    msgpack_object root;
+    msgpack_object error_map;
+    msgpack_object details_arr;
+    msgpack_object logEntryErrors_map;
+    ret = flb_pack_json(c->resp.payload, c->resp.payload_size,
+        &buffer, &size, &root_type, NULL);
+    msgpack_unpacked_init(&result);
+    ret = msgpack_unpack_next(&result, buffer, size, &off);
+    if (ret != MSGPACK_UNPACK_SUCCESS) {
+        flb_plg_error(ctx->ins, "Cannot unpack %s response\n%s", c->resp.payload);
+        flb_free(buffer);
+        msgpack_unpacked_destroy(&result);
+        return -1;
+    }
+
+    root = result.data;
+    if (root.type != MSGPACK_OBJECT_MAP) {
+        flb_plg_error(ctx->ins, "%s response parsing failed, msgpack_type=%i", root.type);
+        flb_free(buffer);
+        msgpack_unpacked_destroy(&result);
+        return -1;
+    }
+
+    ret = extract_map_from_msgpack(&root, "error", 5, MSGPACK_OBJECT_MAP ,&error_map);
```

Comment:
It might help to include an example of what the response looks like (or a link to a doc somewhere) so that this parsing logic is easier to follow.